qa/L0_secure_grpc/test.sh

```diff
@@ -90,15 +94,72 @@ for CASE in server mutual both; do
     set +e
 
     # Test basic inference using grpc secure channel
-    $TEST_CLIENT_PY -v --ssl --root-certificates ca.crt --private-key client.key --certificate-chain client.crt >> ${CLIENT_LOG}.ssl_infer 2>&1
+    $TEST_CLIENT_PY -v --ssl --root-certificates ca.crt --private-key client.key --certificate-chain client.crt >> ${CLIENT_LOG}.${CASE}.ssl_infer 2>&1
+    if [ $? -ne 0 ]; then
+        cat ${CLIENT_LOG}.${CASE}.ssl_infer
+        RET=1
+    fi
+
+    $TEST_CLIENT -v --ssl --root-certificates ca.crt --private-key client.key --certificate-chain client.crt >> ${CLIENT_LOG}.${CASE}.c++.ssl_infer 2>&1
+    if [ $? -ne 0 ]; then
+        cat ${CLIENT_LOG}.${CASE}.c++.ssl_infer
+        RET=1
+    fi
+
+    set -e
+
+    kill $SERVER_PID
+    wait $SERVER_PID
+done
+
+# Test failure cases for SSL
+for CASE in server mutual; do
+    if [ "$CASE" == "server" ]; then
+        SERVER_ARGS="$SERVER_BASE_ARGS --grpc-use-ssl=1"
+    elif [ "$CASE" == "mutual" ]; then
+        SERVER_ARGS="$SERVER_BASE_ARGS --grpc-use-ssl-mutual=1"
+    fi
+
+    run_server
+    if [ "$SERVER_PID" == "0" ]; then
+        echo -e "\n***\n*** Failed to start $SERVER\n***"
+        cat $SERVER_LOG
+        exit 1
+    fi
+
+    set +e
+
+    # Test inference client using grpc secure channel without ssl
+    $TEST_CLIENT_PY -v >> ${CLIENT_LOG}.${CASE}.no_ssl_fail_infer 2>&1
     if [ $? -ne 0 ]; then
-        cat ${CLIENT_LOG}.ssl_infer
+        cat ${CLIENT_LOG}.${CASE}.no_ssl_fail_infer
+        echo -e "\n***\n*** Expected test failure\n***"
+    else
         RET=1
     fi
 
-    $TEST_CLIENT -v --ssl --root-certificates ca.crt --private-key client.key --certificate-chain client.crt >> ${CLIENT_LOG}.c++.ssl_infer 2>&1
+    $TEST_CLIENT -v >> ${CLIENT_LOG}.${CASE}.no_ssl_fail_infer 2>&1
     if [ $? -ne 0 ]; then
-        cat ${CLIENT_LOG}.c++.ssl_infer
+        cat ${CLIENT_LOG}.${CASE}.c++.no_ssl_fail_infer
```

Comment:
The CI is showing a couple of cases where the "cat" is failing because the filename is not correct. After fixing you can just test locally, no need to rerun the entire CI

Reply:
Fixed with last commit